lib/checklist_syntax_migrator.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class ChecklistSyntaxMigrator
+  REGEX = /^ {0,3}\[(_|-|\*|\\\*)\]/
+  OLD_CHARACTERS = /(_|-|\*|\\\*)/
+
+  def initialize(post)
+    @post = post
+  end
+
+  def update_syntax!
+    lines =  @post.raw.split("\n")
+    lines.each_with_index do |line, index|
+      lines[index] = line.gsub(REGEX) {
+        Regexp.last_match(0).gsub(OLD_CHARACTERS, "x")
```

Comment:
The reason for having both these regex's instead of one, is to retain the optional whitespace before the checklists.